npms/skale-ima/index.js

```diff
@@ -5133,14 +5197,12 @@ async function do_transfer(
     let nIdxCurrentMsg = 0;
     let nOutMsgCnt = 0;
     let nIncMsgCnt = 0;
-    let idxLastToPopNotIncluding = 0;
     try {
-        let nPossibleIntegerValue = 0;
         details.write( cc.info( "SRC " ) + cc.sunny( "MessageProxy" ) + cc.info( " address is....." ) + cc.bright( jo_message_proxy_src.options.address ) + "\n" );
         details.write( cc.info( "DST " ) + cc.sunny( "MessageProxy" ) + cc.info( " address is....." ) + cc.bright( jo_message_proxy_dst.options.address ) + "\n" );
         strActionName = "src-chain.MessageProxy.getOutgoingMessagesCounter()";
         details.write( strLogPrefix + cc.debug( "Will call " ) + cc.notice( strActionName ) + cc.debug( "..." ) + "\n" );
-        nPossibleIntegerValue = await jo_message_proxy_src.methods.getOutgoingMessagesCounter( chain_id_dst ).call( {
+        let nPossibleIntegerValue = await jo_message_proxy_src.methods.getOutgoingMessagesCounter( chain_id_dst ).call( {
```

Comment:
This call will fail if smart contracts have older version.
The optimization must not be applied if communication with non updated contracts.

Reply:
will need to do more commits to this PR

Reply:
fixed

Reply:
fixed